src/ml_board/backend/restful_api/data_access.py

```diff
@@ -96,6 +110,12 @@ def iterfile(file_path: str):
         with open(file_path, mode="rb") as file_like:
             yield from file_like
 
+    @staticmethod
+    def iterbytefile(file_path: str):
+        with open(file_path, mode="rb") as f:
+            buffer = BytesIO(f.read())
+        return buffer
+
```

Comment:
The most common use of this API is during warm start, the main application is in the mlgym core so since we need a bytes response from the API, I do not see us having any other choice. The frontend use of downloading checkpoint is rare I guess, the only fix I can see is to add another parameter named "client_use" = True / False, if true then we use the generator to send a response to the front end if false then we load the entire thing as in the code. What do you think?

Reply:
Yes, that is possible but then we need to first get the data from the Generator type response from the API by reading the file and then convert it into BytesIO type to be then loaded for the warm start. The only reason I made the change was to be able to load the model.pickle which I was not able to do when I was returning the Generator response from the API. The only way it would work is if I returned BytesIO() response. Could you create an issue for this. I can check it again and try to come up with a solution.

Reply:
It's definitely possible. You can just return the file within FastAPI as documented here: 
https://fastapi.tiangolo.com/advanced/custom-response/?h=fileresponse#fileresponse

The mlgym backend can just push the incoming data into a bytesIO stream.